server/src/main/java/org/elasticsearch/common/util/BigDoubleArray.java

```diff
@@ -123,4 +131,21 @@ public static long estimateRamBytes(final long size) {
     public void set(long index, byte[] buf, int offset, int len) {
         set(index, buf, offset, len, pages, 3);
     }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        int size = (int) this.size;
+        out.writeVInt(size * Double.BYTES);
+        int lastPageEnd = size % DOUBLE_PAGE_SIZE;
+        if (lastPageEnd == 0) {
+            for (byte[] page : pages) {
+                out.write(page);
+            }
+            return;
+        }
+        for (int i = 0; i < pages.length - 1; i++) {
+            out.write(pages[i]);
+        }
+        out.write(pages[pages.length - 1], 0, lastPageEnd * Double.BYTES);
```

Comment:
I wonder if we should share this code with the int/long/whatever other types. It's nearly the same code. :shrug: sounds like a good follow up change.